src/sv_primlit_integral.rs

```diff
@@ -50,50 +50,65 @@ impl SvPrimaryLiteralIntegral {
 
             for _x in 0..diff {
                 right_nu.data_01.insert(0, 0);
+                if right_nu.is_4state() {
+                    right_nu.data_xz.as_mut().unwrap().insert(0, 0);
+                }
             }
         } else if left_size < right_size {
             let diff: usize = right_size - left_size;
 
             for _x in 0..diff {
                 self.data_01.insert(0, 0);
+                if self.is_4state() {
+                    self.data_xz.as_mut().unwrap().insert(0, 0);
+                }
             }
         }
-
-        if self.is_4state() {
-            self.data_xz = self.to_4state().data_xz;
-        }
-
-        if right_nu.is_4state() {
-            right_nu.data_xz = right_nu.to_4state().data_xz;
-        }
     }
 
     /* Receives a signed integral primary literal as an argument and deduces whether the stored value is -ve or +ve based on the size value set. */
     pub fn is_negative(&mut self) -> bool {
         if self.signed != true {
             panic!("Expected signed SvPrimaryLiteralIntegral but found unsigned!");
```

Comment:
What about something like this to make it accept unsigned numbers?
```rust
if !self.signed {
    false
} else {
    let zero = ...;
    let one = ...;
    one == self.lt(zero) // Do you need to clone() zero? Maybe borrow a ref instead?
}
```

Reply:
By default the branch checking whether the prim lit is signed is not needed. I.e smth_unsigned.lt(zero) would return zero. I have just added it to restrict the use of is_negative to signed only and give an error.

Reply:
Ah okay. Is it sensible to be unrestricted? I'm thinking for future use where a dev will expect that function to be usable.

Reply:
So far it was an additional measure to ensure that the signed and unsigned distinction works as expected. But since no errors have been caught so far, I believe that the distinction works correctly. I'm will have it removed in the next commit,